tests/test_pytest.py

```diff
@@ -399,3 +399,21 @@ def collect_names(code):
         ) == {"Test","_","a", "self", "__thing"}
 
 
+from executing.executing import is_rewritten_by_pytest,get_instructions
+import dis
+
+def test_pytest_rewrite():
+    frame=inspect.currentframe()
+    #dis.dis(frame.f_code)
+    for bc in get_instructions(frame.f_code):
+        print(bc)
+    
+    # check for assert statements rewrite caused by this assert
+    assert is_rewritten_by_pytest(frame.f_code)==True
+
+def test_no_pytest_rewrite():
+    frame=inspect.currentframe()
+
+    # no assert -> no rewrite
+    if is_rewritten_by_pytest(frame.f_code):
+        raise AssertionError("unexpected pytest assert rewrite")
```

Comment:
```suggestion
    assert not is_rewritten_by_pytest(frame.f_code)
```

Reply:
This will not work. Pytest will rewrite the assert and the test will fail. 

The new check checks only for rewritten assertions in the function and not in the module. Is this ok for the SentinelParser?

Reply:
> This will not work. Pytest will rewrite the assert and the test will fail.

lol good point

> The new check checks only for rewritten assertions in the function and not in the module. Is this ok for the SentinelParser?

Yes, that should be OK, and that should be the same as the previous behavious since that also only looked at the current frame's code object.